src/ComboBox/ComboBox.svelte

```diff
@@ -299,11 +300,33 @@
                 selectedId = filteredItems[highlightedIndex].id;
               }
             } else {
-              open = false;
-              if (filteredItems[0]) {
-                value = itemToString(filteredItems[0]);
-                selectedItem = filteredItems[0];
-                selectedId = filteredItems[0].id;
+              // searching typed value in text list with lowercase
+              /** @type Array<ComboBoxItem> */
+              const listOfItems = filteredItems.map((e) => {
+                return {
+                  id: e.id,
+                  text: e.text?.toLowerCase(),
+                  disabled: e.disabled,
+                };
+              });
+              const matchedItem = listOfItems.find(
+                (e) => e.text === value?.toLowerCase()
+              );
+              if (matchedItem) {
+                // typed value has matched
+                open = false;
+                selectedItem = filteredItems.find(
+                  (e) => e.id === matchedItem.id
+                );
+                value = selectedItem.text;
+                selectedId = selectedItem.id;
+              } else {
+                open = false;
+                if (filteredItems[0]) {
+                  value = itemToString(filteredItems[0]);
+                  selectedItem = filteredItems[0];
+                  selectedId = filteredItems[0].id;
+                }
```

Comment:
Would try to avoid duplicating the index.
Also, this probably should `find` the first non-disabled item instead of just picking index `0`.

Reply:
OK, I think that's a good suggestion. This will be a separate commit.